core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/utils/ConnObjectUtils.java

```diff
@@ -266,7 +266,6 @@ public RealmTO getRealmTO(final ConnectorObject obj, final PullTask task, final
                     }
 
                     updatedUser.setSecurityQuestion(updatedUser.getSecurityQuestion());
-                    updatedUser.setMustChangePassword(originalUser.isMustChangePassword());
```

Comment:
Why is this line removed? Are you sure this works either when `mustChangePassword` is mapped and when is not mapped?

Reply:
I will check this

Reply:
1. this line always adds the value to updatedUser from originalUser. It means that `mustChangePassword` flag will never be updated during pull operation.
2. Also, it is removed to correctly calculate the UserPatch. Otherwise, `org.apache.syncope.common.lib.patch.UserPatch#mustChangePassword` is calculated incorrectly.

Reply:
I agree that the current line is not correct, but the actual fix is to just remove it, the actual fix is to avoid that line in case `mustChangePassword` is mapped.

In order to do so, I would propose the following changes:

* move `UserDataBinderImpl#isMustChangePwdMappingItemConfigured` to `MappingUtils` as follows:
```java
    public static boolean hasMustChangePassword(final Provision provision) {
        return provision != null && provision.getMapping() != null
                && IterableUtils.matchesAny(provision.getMapping().getItems(), new Predicate<MappingItem>() {

                    @Override
                    public boolean evaluate(final MappingItem mappingItem) {
                        return "mustChangePassword".equals(mappingItem.getIntAttrName());
                    }
                });
    }
```
* adjust (and simplify) the code you added in `UserDataBinderImpl#update`
* in `ConnObjectUtils`, replace
```java
updatedUser.setMustChangePassword(originalUser.isMustChangePassword());
```
with
```java
if (!MappingUtils.hasMustChangePassword(provision)) {
  updatedUser.setMustChangePassword(originalUser.isMustChangePassword());
}
```

WDYT?

Reply:
Yes, you're right. If mustChangePassword is not mapped and originalUser has mustChangePwd field set to true, than the value will be overwritten and set to `false`. Agree with you. Will apply your code.

Reply:
updated the code as you suggested